packages/expo-cli/src/commands/eas-build/build/builders/AndroidBuilder.ts

```diff
@@ -81,68 +88,70 @@ class AndroidBuilder implements Builder<Platform.Android> {
     const buildGradleContent = await fs.readFile(path.join(buildGradlePath), 'utf-8');
     const applyEasGradle = 'apply from: "./eas-build.gradle"';
 
-    const hasEasGradleApply = buildGradleContent
-      .split('\n')
-      // Check for both single and double quotes
-      .some(line => line === applyEasGradle || line === applyEasGradle.replace(/"/g, "'"));
+    const hasEasGradleApply = hasApplyLine(buildGradleContent, applyEasGradle);
+    const isUpdatesConfigured = await isUpdatesConfiguredAndroidAsync(
+      this.ctx.commandCtx.projectDir
+    );
 
-    return hasEasGradleApply && hasEasGradleFile;
+    return hasEasGradleApply && hasEasGradleFile && isUpdatesConfigured;
```

Comment:
There should be a better error message if only updates are not configured

inside `setUpdatesVersionsAndroidAsync` check `isUpdatesConfiguredAndroidAsync` and throw exception if not configured